roles/tests-integration/tests/common/sniffer.rs

```diff
@@ -0,0 +1,475 @@
+                }
+                IdentifyTransactions(m) => {
+                    AnyMessage::JobDeclaration(IdentifyTransactions(m.into_static()))
+                }
+                IdentifyTransactionsSuccess(m) => {
+                    AnyMessage::JobDeclaration(IdentifyTransactionsSuccess(m.into_static()))
+                }
+                ProvideMissingTransactions(m) => {
+                    AnyMessage::JobDeclaration(ProvideMissingTransactions(m.into_static()))
+                }
+                ProvideMissingTransactionsSuccess(m) => {
+                    AnyMessage::JobDeclaration(ProvideMissingTransactionsSuccess(m.into_static()))
+                }
+                SubmitSolution(m) => AnyMessage::JobDeclaration(SubmitSolution(m.into_static())),
+            },
+            AnyMessage::TemplateDistribution(m) => match m {
+                CoinbaseOutputDataSize(m) => {
+                    AnyMessage::TemplateDistribution(CoinbaseOutputDataSize(m.into_static()))
+                }
+                TemplateDistribution::NewTemplate(m) => AnyMessage::TemplateDistribution(
+                    TemplateDistribution::NewTemplate(m.into_static()),
+                ),
+                TemplateDistribution::RequestTransactionData(m) => {
+                    AnyMessage::TemplateDistribution(TemplateDistribution::RequestTransactionData(
+                        m.into_static(),
+                    ))
+                }
+                TemplateDistribution::RequestTransactionDataError(m) => {
+                    AnyMessage::TemplateDistribution(
+                        TemplateDistribution::RequestTransactionDataError(m.into_static()),
+                    )
+                }
+                TemplateDistribution::RequestTransactionDataSuccess(m) => {
+                    AnyMessage::TemplateDistribution(
+                        TemplateDistribution::RequestTransactionDataSuccess(m.into_static()),
+                    )
+                }
+                TemplateDistribution::SetNewPrevHash(m) => AnyMessage::TemplateDistribution(
+                    TemplateDistribution::SetNewPrevHash(m.into_static()),
+                ),
+                TemplateDistribution::SubmitSolution(m) => AnyMessage::TemplateDistribution(
+                    TemplateDistribution::SubmitSolution(m.into_static()),
+                ),
+            },
+        }
+    }
+
+    async fn wait_for_client(client: SocketAddr) -> TcpStream {
+        let listner = TcpListener::bind(client)
+            .await
+            .expect("Impossible to listen on given address");
+        if let Ok((stream, _)) = listner.accept().await {
+            stream
+        } else {
+            panic!("Impossible to accept dowsntream connection")
+        }
+    }
+}
+
+// Utility macro to assert that the downstream and upstream roles have sent specific messages.
+#[macro_export]
+macro_rules! assert_message {
```

Comment:
How does this differ from the other macros? 
I tried to change the test to use this one (to assert the `SetupConnectionSuccess`) and I was getting some compilation errors. So I think I was not using it in the right way? Maybe is worth editing the test to assert one message using this macro?
I'm sure I'm missing something